src/lazy/expanded/compiler.rs

```diff
@@ -602,15 +712,11 @@ impl TemplateCompiler {
         let mut expressions = lazy_sexp.iter();
         // Convert the macro ID (name or address) into an address. If this refers to a macro that
         // doesn't exist yet, this will return an error. This prevents recursion.
-        // TODO: Consider storing the name of the invoked target macro in the host's definition
-        //       as debug information. The name cannot be stored directly on the
-        //       TemplateBodyMacroInvocation as that would prevent the type from being `Copy`.
-        let (_maybe_name, macro_address) =
-            Self::name_and_address_from_id_expr(tdl_context, expressions.next())?;
+        let macro_ref = Self::name_and_address_from_id_expr(tdl_context, expressions.next())?;
         let macro_step_index = definition.expressions.len();
         // Assume the macro contains zero argument expressions to start, we'll update
         // this at the end of the function.
-        definition.push_macro_invocation(macro_address, ExprRange::empty());
+        definition.push_macro_invocation(Rc::clone(&macro_ref), ExprRange::empty());
```

Comment:
🗺️ `Rc`s are only _created_ when the `Macro` is compiled. They might get cloned (i.e. shallow-copied) some number of times after that during compilation. After compilation, however, only `&` references are used. This should mean that `Rc` overhead is negligible when amortized across the lifetime of the stream.